tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_activation_z_score_threshold.ipynb

```diff
@@ -0,0 +1,809 @@
+    "!pip install -q torch torchvision"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "bd8e08612add2018",
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "import importlib\n",
+    "if not importlib.util.find_spec('model_compression_toolkit'):\n",
+    "    !pip install model_compression_toolkit"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "b3f0acc8-281c-4bca-b0b9-3d7677105f19",
+   "metadata": {
+    "id": "b3f0acc8-281c-4bca-b0b9-3d7677105f19"
+   },
+   "outputs": [],
+   "source": [
+    "import torch\n",
+    "from torch.utils.data import DataLoader\n",
+    "from torchvision.models import mobilenet_v2, MobileNet_V2_Weights\n",
+    "from torchvision.datasets import ImageNet"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fd5ac404451fd924",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "Load a pre-trained MobileNetV2 model from Keras, in 32-bits floating-point precision format."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "69daa2d5d731b157",
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "weights = MobileNet_V2_Weights.IMAGENET1K_V2\n",
+    "\n",
+    "float_model = mobilenet_v2(weights=weights)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a5584696d8f09653",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "## Dataset preparation\n",
+    "### Download the ImageNet validation set\n",
+    "Download the ImageNet dataset with only the validation split.\n",
+    "**Note:** For demonstration purposes we use the validation set for the model quantization routines. Usually, a subset of the training dataset is used, but loading it is a heavy procedure that is unnecessary for the sake of this demonstration.\n",
+    "\n",
+    "This step may take several minutes..."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "66a1e4f3878aa76b",
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    " \n",
+    "if not os.path.isdir('imagenet'):\n",
+    "    !mkdir imagenet\n",
+    "    !wget -P imagenet https://image-net.org/data/ILSVRC/2012/ILSVRC2012_devkit_t12.tar.gz\n",
+    "    !wget -P imagenet https://image-net.org/data/ILSVRC/2012/ILSVRC2012_img_val.tar"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "bddd52741649281e",
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "from pathlib import Path\n",
+    "import shutil\n",
+    "\n",
+    "root = Path('./imagenet')\n",
+    "imgs_dir = root / 'ILSVRC2012_img_val'\n",
+    "target_dir = root /'val'\n",
+    "\n",
+    "def extract_labels():\n",
+    "    !pip install -q scipy\n",
+    "    import scipy\n",
+    "    mat = scipy.io.loadmat(root / 'ILSVRC2012_devkit_t12/data/meta.mat', squeeze_me=True)\n",
```

Comment:
For some reason, this line fails when I try to run the tutorial. I get this error: "OSError: Reader needs file name or open file-like object". Can you please check for the reason?

Reply:
https://colab.research.google.com/github/gouda-youichi/model_optimization/blob/adding_example_pytorch_activation_z_score_threshold/tutorials/notebooks/mct_features_notebooks/pytorch/example_pytorch_activation_z_score_threshold.ipynb